python/ray/dag/compiled_dag_node.py

```diff
@@ -153,7 +153,14 @@ def __init__(self, idx: int, dag_node: "ray.dag.DAGNode"):
         self.dag_node = dag_node
 
         self.downstream_node_idxs: Dict[int, "ray.actor.ActorHandle"] = {}
-        self.output_channel = None
+        # Multiple return values are written to separate `output_channels`.
+        # `output_idxs` represents the tuple index of the output value for
+        # multiple returns in a tuple. If an output index is None, it means
+        # the complete return value is written to the output channel.
+        # Otherwise, the return value is a tuple and the index is used
+        # to extract the value to be written to the output channel.
+        self.output_channels: List[ChannelInterface] = []
+        self.output_idxs: List[Optional[int]] = []
```

Comment:
Is there a way to encapsulate these? e.g., something like `DAGInputAdapter`

Reply:
For now, I think it is fine to keep these.